lib/tdf3/src/models/encryption-information.ts

```diff
@@ -137,9 +139,10 @@ export class SplitKey {
   }
 
   async write(policy: Policy, keyInfo: KeyInfo): Promise<EncryptionInformation> {
-    const algorithm = this.cipher.name;
+    const algorithm = this.cipher?.name;
     if (!algorithm) {
-      throw new Error('Uninitialized cipher type');
+      // Hard coded as part of the cipher object. This should not be reachable.
+      throw new Error('internal: uninitialized cipher type');
```

Comment:
base error?

Reply:
While I think this is most likely an internal error, it could conceivably also be a configuration error if the client is highly customized (a la secure share)